safe-svg.php

```diff
@@ -486,39 +488,26 @@ protected function svg_dimensions( $svg ) {
         }
 
         /**
-         * Fix the output of images using wp_get_attachment_image
+         * Disable the creation of srcset on SVG images.
          *
-         * @param array $attr Attributes for the image markup.
-         * @param WP_Post $attachment Image attachment post.
-         * @param string|array $size Requested size. Image size or array of width and height values
-         *                                 (in that order). Default 'thumbnail'.
+         * @param array $image_meta The image meta data.
+         * @param int[]  $size_array    {
+         *     An array of requested width and height values.
+         *
+         *     @type int $0 The width in pixels.
+         *     @type int $1 The height in pixels.
+         * }
+         * @param string $image_src     The 'src' of the image.
+         * @param int    $attachment_id The image attachment ID.
          */
-        public function fix_direct_image_output( $attr, $attachment, $size = 'thumbnail' ) {
-
-            // If we're not getting a WP_Post object, bail early.
-            // @see https://wordpress.org/support/topic/notice-trying-to-get-property-id/
-            if ( ! $attachment instanceof WP_Post ) {
-                return $attr;
+        public function disable_srcset( $image_meta, $size_array, $image_src, $attachment_id ) {
+            if ( $attachment_id && 'image/svg+xml' === get_post_mime_type( $attachment_id ) ) {
+                $image_meta['sizes'] = array();
```

Comment:
The root cause of the wrong srcset image sizes is that WordPress core function `image_downsize()` always returns square size for SVG images, even if they are rectangular.

We could preempt it with `image_downsize` filter which will perform custom calculations of resized SVG width and height.

Reply:
> We could preempt it with `image_downsize` filter which will perform custom calculations of resized SVG width and height

Are you suggesting we use that filter and return just the full SVG src?

I guess my opinion here is we don't ever need `srcset` or `sizes` for SVG images, since we will only ever have a single SVG file and not have resized versions of that, like WordPress will do for normal images. Unless I'm missing something, seems pointless to ever have those attributes as it will just always serve the exact same image no matter the breakpoint